keps/sig-network/2079-network-policy-port-range/README.md

```diff
@@ -0,0 +1,623 @@
+-->
+
+API changes to NetworkPolicy:
+* Add a new field ``spec.ingress|egress.ports.PortRange
+* On field creation/change a validation should happen to verify if the format is compliant with ``nnnnn,nnnn1-nnnn5,http`` where nnnnn is a number between 0 and 65535 and named ports aren't mixed with numbered ports
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. The KEP
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+-->
+#### Alpha 
+- Add a feature gated new field to NetworkPolicy
+- Communicate CNI providers about the new field
+- Add validation tests in API
+
+#### Beta
+- ``PortRanges`` has been supported for at least 1 minor release
+- At least one CNI provider implements the new field
+- Feature Gate is enabled by Default
+
+#### GA Graduation
+
+- At least two CNIs support the ``PortRanges`` field
+- ``PortRanges`` has been enabled by default for at least 1 minor release
+
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a Deprecated Flag
```

Comment:
You should get rid of as much template text as you can to make it easier to see the actual KEP text. eg, just delete any sections (like this one about command-line flags) which are irrelevant to your KEP.